napari/layers/image/_tests/test_image.py

```diff
@@ -667,6 +667,10 @@ def test_world_data_extent():
     layer = Image(data)
     extent = np.array(((0,) * 3, shape))
     check_layer_world_data_extent(layer, extent, (3, 1, 1), (10, 20, 5), True)
+    extent2 = layer.extent
```

Comment:
somthing like now in this test?

Reply:
I was thinking something that checks that the extent value changes (not just the identity), like the following.

```python
before = layer.extent
layer.scale = (2, 2, 2)
assert before != layer.extent
```

It might be better to check that the new `layer.extent` has the expected value, but that's actually tricky to reason about because `check_layer_world_data_extent` mutates `layer`.

Alternatively (and probably my most preferred solution) is to isolate out this specific behavior into a separate test.

```python
def test_set_scale_updates_extent():
     layer = Image(np.zeros((3, 4), scale=(1, 1))
     extent = layer.extent

     layer.scale = (2, 1)

     assert layer.extent != extent
```

Reply:
this test does not check if the extent is cached. It will pass with code from the main branch.